### ISSUE TITLE
Handoff between real-time and historical processing

### ISSUE THREAD

**Issue by Ishatt** [near/queryapi#209]
Description

Problem:

Currently we do not coordinate between real-time and historical processing, indexers can run on blocks out of order. 

See Confluence pages

See [Real-time to Historical communication](https://pagodaplatform.atlassian.net/wiki/spaces/DAT/pages/333545513/Real-time+to+Historical+communication) for architectural concerns.

Proposed Solutions:

Append real-time blocks to be processed to the processing queue of the historical thread.

After the move to Redis we could store a redis flag for each function for historical_backfill_in_progress . When this is set at the start of historical backfill, real-time processing can skip processing until both the flag is cleared and the historical queue is empty. Once the historical queue has been filled the flag can be cleared.
```[tasklist]
### Blocked by
- [ ] https://github.com/near/queryapi/issues/198
```

**Comment by pkudinov** [comment]
this is a duplicate of #217
